maven-extension/src/main/java/io/opentelemetry/maven/handler/MavenDeployHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.maven.handler;
+
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.maven.MavenGoal;
+import io.opentelemetry.maven.semconv.MavenOtelSemanticAttributes;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.artifact.repository.Authentication;
+import org.apache.maven.execution.ExecutionEvent;
+import org.apache.maven.project.MavenProject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Note: Later, we may prefer to have one span per actual upload than a generic span. We could
+ * achieve this instrumenting the <a
+ * href="https://projects.eclipse.org/projects/technology.aether">Aether library</a>
+ */
+public class MavenDeployHandler extends AbstractMojoGoalExecutionHandler
+    implements MojoGoalExecutionHandler {
+  private static final Logger logger = LoggerFactory.getLogger(MavenDeployHandler.class);
+
+  @Override
+  public void enrichSpan(SpanBuilder spanBuilder, ExecutionEvent execution) {
+    spanBuilder.setSpanKind(SpanKind.CLIENT);
+
+    MavenProject project = execution.getProject();
+    Optional<ArtifactRepository> optRepository =
+        Optional.ofNullable(project.getDistributionManagementArtifactRepository());
+
+    optRepository.ifPresent(
```

Comment:
For example, `ofNullable` followed `ifPresent` like this is almost exactly the same as normal null checks, but adds verbose variables like `optRepository` and I think some extra indent too

Reply:
Migration from `Optional` to `@Nullable` done